manager/src/main/java/org/openremote/manager/map/MapResourceImpl.java

```diff
@@ -71,4 +76,44 @@ public byte[] getTile(int zoom, int column, int row) {
             throw new WebApplicationException(Response.Status.NO_CONTENT);
         }
     }
+
+    @Override
+    public Response uploadMap(@Context HttpServletRequest request) {
+        if (request.getContentLength() > mapService.customMapLimit) {
+            throw new WebApplicationException("{\"map-custom\": false}", Response.Status.REQUEST_ENTITY_TOO_LARGE);
+        }
+
+        try (InputStream stream = request.getInputStream()) {
+            boolean isSaved = mapService.saveUploadedFile(stream);
+            ObjectNode response = ValueUtil.JSON
+                .createObjectNode()
+                .put("map-custom", isSaved);
+
+            if (isSaved) {
+                return Response.ok("File uploaded successfully").build();
+            }
+            return Response.status(Response.Status.BAD_REQUEST).entity(response).build();
+        } catch (IOException error) {
+            throw new WebApplicationException("{\"map-custom\": false}", Response.Status.BAD_REQUEST);
+        }
```

Comment:
Don't see what value the `response` object have and then in the catch you manually type out the JSON, HTTP status code is sufficient here but please log details of exceptions or known errors.